src/ProductDetail.jsx

```diff
@@ -4,29 +4,28 @@ import "./OneEntry.css"
 import { Box } from "./components/Box";
 import { Text } from "./components/Text";
 
-const OneEntry = () =>{
-    const id = useParams()
+const ProductDetail = () =>{
+    const {id} = useParams()
     // const id = props.id
     const {data:entry, loading, error} = useFetch("products/" + id)
-    console.log(entry, "entry")
 
     return(
         <>
-            {/* <h1 style={{color:"white"}}>The ID is {id}</h1> */}
-            <Box>
+            <Box detail>
                 <img src={`${entry.image}`} className="prodimg2" />
                 <div>
                     <Text color="navy" size="25px" link linkSrc={`${entry.id}`}>
                         {entry.title}
                     </Text>
-                    <Text color="dimgray" size="20px">{entry.category} - {entry.rating.rate} stars ({entry.rating.count} ratings)</Text>
+                    {/* <Text color="dimgray" size="20px">{entry.category} - {entry.rating.rate} stars ({entry.rating.count} ratings)</Text> */}
```

Comment:
This error from your browser console sheds some light on what's going on.

<img width="591" alt="Screenshot 2023-05-07 at 10 22 14 AM" src="https://user-images.githubusercontent.com/46625943/236683335-53d573e2-2ebb-40ff-a8ee-471a9df13ae2.png">

If you had a loading state on this screen you wouldn't see this error, and here's why:

In `ProductDetail`,  we are waiting for an **asynchronous** API call to be completed. As soon as `ProductDetail` is rendered, it tries to read the `entry` object before your API request is completed. So on render, `entry === []`. 

Javascript **will not** throw an error when trying to read a property on the array, or in our case `entry.title`, JS will just say it's undefined.
```
const entry = []
console.log(entry.title) // undefined
```

Javascript **will** throw an error when trying to read a property from `undefined` or in our case `entry.rating.rate`.
```
const entry = []
console.log(entry.title.anyPropertyName) // Error
```

By adding the loading state, we return early and only fall through to our product detail JSX when loading is false **and** we have data.